src/reference/asciidoc/delayer.adoc

```diff
@@ -187,3 +189,20 @@ Message<String> delayerReschedulingMessage =
 ====
 
 NOTE: For more information regarding the message store, JMX, and the control bus, see <<system-management-chapter>>.
+
+[[delayer-release-failures]]
+==== Release Failures
+
+Starting with version 5.0.8, there are two new properties on the delayer:
+
+- `maxAttempts` (default 5)
+- `retryDelay` (default 1 second)
+
+When a message is released, if the downstream flow fails, the release will be attempted after the `retryDelay`.
+If the `maxAttempts` is reached, the message is discarded (unless the release is transactional, in which case the message will remain in the store, but will no longer be scheduled for release, until the application is restarted).
```

Comment:
Well, we always should not forget that `DelayHandler` is a `@ManagedResource` and has these management operations:
```
public interface DelayHandlerManagement {

	@ManagedAttribute
	int getDelayedMessageCount();

	@ManagedOperation
	void reschedulePersistedMessages();
}
```

So, with the Control Bus command we always can reschedule at runtime.

Reply:
Oh, dear; I think there's a problem with that - `releaseMessageAfterDelay()` should only be invoked when this is called after the context refresh. For a managed operation, we should only release past-due messages; otherwise future messages will be duplicated. And, it's not that simple; there will be race conditions.

But, I think resolving this is outside of this PR.

Reply:
Ah, wait; we're protected against this by `removeDelayedMessageFromMessageStore`. We could consider optimizing, though, to avoid double-scheduling future events.